src/main/java/seedu/staffsnap/model/applicant/CustomFilterPredicate.java

```diff
@@ -0,0 +1,123 @@
+package seedu.staffsnap.model.applicant;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import seedu.staffsnap.commons.util.StringUtil;
+import seedu.staffsnap.commons.util.ToStringBuilder;
+import seedu.staffsnap.model.interview.Interview;
+
+
+/**
+ * Custom predicate to be used to filter applicants
+ */
+public class CustomFilterPredicate implements Predicate<Applicant> {
+
+    // Identity fields
+    private final Name name;
+    private final Phone phone;
+    // Data fields
+    private final Email email;
+    private final Position position;
+    private final List<Interview> interviews;
+    private final Status status;
+    private final Double lessThanScore;
+    private final Double greaterThanScore;
+
+    /**
+     * Constructor for CustomFilterPredicate
+     *
+     * @param name       Name of applicant
+     * @param phone      Phone number of applicant
+     * @param email      Email address of applicant
+     * @param position   Position applied for by applicant
+     * @param interviews Interviews applicant has to go through
+     */
+    public CustomFilterPredicate(Name name, Phone phone, Email email, Position position, List<Interview> interviews,
+                                 Status status, Double lessThanScore, Double greaterThanScore) {
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.position = position;
+        this.interviews = interviews;
+        this.status = status;
+        this.lessThanScore = lessThanScore;
+        this.greaterThanScore = greaterThanScore;
+    }
+
+    /**
+     * @param applicant the input argument
```

Comment:
I feel that "the input argument" doesn't seem to add much value here, perhaps better to change to "the applicant to be tested"? 🤔